asdf/commands/edit.py

```diff
@@ -0,0 +1,589 @@
+"""
+Contains commands for lightweight text editing of an ASDF file.
+Future work: Make this interactive editing.
+"""
+
+import io
+import os
+import struct
+import sys
+
+import asdf.constants as constants
+
+from .. import generic_io
+from .. import schema
+from .. import yamlutil
+
+from .main import Command
+
+__all__ = ["edit"]
+
+
+class Edit(Command):
+    @classmethod
+    def setup_arguments(cls, subparsers):
+        """
+        Set up a command line argument parser for the edit subcommand.
+        """
+        desc_string = (
+            "Allows for easy editing of the YAML in an ASDF file.  "
+            "For edit mode, the YAML portion of an ASDF file is"
+            "separated from the ASDF into a text file for easy"
+            "editing.  For save mode, the edited text file is written"
+            "to its ASDF file."
+        )
+
+        # Set up the parser
+        parser = subparsers.add_parser(
+            str("edit"),
+            help="Edit YAML portion of an ASDF file.",
+            description=desc_string,
+        )
+
+        # Need an input file
+        parser.add_argument(
+            "--infile",
+            "-f",
+            type=str,
+            required=True,
+            dest="fname",
+            help="Input file (ASDF for -e option, YAML for -s option",
+        )
+
+        # Need an output file
+        parser.add_argument(
+            "--outfile",
+            "-o",
+            type=str,
+            required=True,
+            dest="oname",
+            help="Output file (YAML for -e option, ASDF for -s option",
+        )
+
+        # The edit is either being performed or saved
+        group = parser.add_mutually_exclusive_group(required=True)
+
+        group.add_argument(
+            "-s",
+            action="store_true",
+            dest="save",
+            help="Saves a YAML text file to an ASDF file.  Requires a "
+            "YAML input file and ASDF output file.",
+        )
+
+        group.add_argument(
+            "-e",
+            action="store_true",
+            dest="edit",
+            help="Create a YAML text file for a ASDF file.  Requires a ASDF input file.",
+        )
+
+        parser.set_defaults(func=cls.run)
+
+        return parser
+
+    @classmethod
+    def run(cls, args):
+        """
+        Execute the edit subcommand.
+        """
+        return edit(args)
+
+
+def is_yaml_file(fname):
+    """
+    Determines if a file is a YAML file based only on the file extension.
+
+    Parameters
+    ----------
+    fname : str
+        Input file name.
+
+    Return
+    ------
+    bool
+    """
+
+    base, ext = os.path.splitext(fname)
+    if ".yaml" != ext:
```

Comment:
Sometimes YAML files also use a `.yml` extension.  Don't know if we will ever come across one of these for this function?

Reply:
I added that extension as a check.